internal/configs/configmaps.go

```diff
@@ -347,10 +347,14 @@ func ParseConfigMap(ctx context.Context, cfgm *v1.ConfigMap, nginxPlus bool, has
 
 	if mainTemplate, exists := cfgm.Data["main-template"]; exists {
 		cfgParams.MainTemplate = &mainTemplate
+	} else {
+		cfgParams.MainTemplate = nil
 	}
 
 	if ingressTemplate, exists := cfgm.Data["ingress-template"]; exists {
 		cfgParams.IngressTemplate = &ingressTemplate
+	} else {
+		cfgParams.IngressTemplate = nil
 	}
```

Comment:
Should we update this for `virtualserver` and `transportserver` also?

Reply:
We may extend this functionality for VS and TS but I'd do this in a separate PR to keep PR's focused and small.